### PR TITLE
fsanitize=undefined,address fixes

### PR DESCRIPTION
**Pull request by Skomski**
Fixes undefined behaviour because memcpy second parameter is defined as nonnull and compiler optimizes based upon that.
Fixes two memory leaks in tests

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -1082,7 +1082,8 @@ int uv_fs_read(uv_loop_t* loop, uv_fs_t* req,
   if (req->bufs == NULL)
     return -ENOMEM;
 
-  memcpy(req->bufs, bufs, nbufs * sizeof(*bufs));
+  if (bufs)
```
Comment (saghul) [inline src/unix/fs.c]: I think we should check if bufs was provided first thing and return UV_EINVAL if it's null, or nbufs is zero.

**Comment by saghul** [comment]
Left a comment about the memcpy approach. The other one LGTM, thanks!

**Comment by Skomski** [comment]
Updated.

**Comment by saghul** [comment]
@Skomski Thanks! Specially for adding an extra test :-) Can you please also make the change on Windows? (There you should return UV_EINVAL instead of -EINVAL)

**Comment by Skomski** [comment]
Updated for win.

**Comment by saghul** [comment]
CI run: https://jenkins-iojs.nodesource.com/view/libuv/job/libuv+any-pr+multi/127/

**Comment by saghul** [comment]
LGTM :+1:

**Comment by saghul** [comment]
Landed in c2e6f3bad1d1e508fa4ce09ba460b398557593d4 and 939ea06f458af6863734c5ebc2fd10a869cc7e5b thanks!

**Pull request closed without merging**